src/system/group.cr

```diff
@@ -0,0 +1,152 @@
+require "c/grp"
+require "c/sys/limits"
+
+struct System::Group
+  {% if flag?(:darwin) || flag?(:openbsd) || flag?(:freebsd) %}
+    GID_MAX = LibC::GID_MAX
+  {% elsif flag?(:linux) %}
+    GID_MAX = 0xffffffff_u32
+  {% else %}
+    GID_MAX = 0xffff_u32
+  {% end %}
+  NGROUPS_MAX = LibC::NGROUPS_MAX
+
+  # Returns the group name for the given group ID.
+  # Raises `System::Group::NotFoundError` if no user exists.
+  def self.name(gid : Int) : String
```

Comment:
You can DRY-it moar (ditto all non-nilable getters below):

```cr
def self.name(gid : Int) : String
  name?(gid) || raise NotFoundError.new("Group with gid #{gid.inspect} was not found")
end
```

Also, please keep exception messages consistent with rest of the code base (ie. with no final `.`).

Reply:
Fixed in all locations in `User` and `Group`.